Fixed #7241 - Some files still use the DB global variable

dfstrauss opened the pull request:
Replaced references to `global $db` with `$db = DBManagerFactory::getInstance()` as per Issue #7241

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Final checklist
- [x] My code follows the code style of this project found [here](https://docs.suitecrm.com/community/contributing-code/coding-standards/).
- [ ] My change requires a change to the documentation.
- [x] I have read the [**How to Contribute**](https://docs.suitecrm.com/community/contributing-code/) guidelines.

dfstrauss commented:
@Dillon-Brown Forgive me my newbie mistakes...
I closed Pull Request 8098 inadvertently because I did not create a dedicated branch on my fork and force-pushed to my `hotfix-7.10.x` branch. This PR applies the same code from my `issue-7241` branch

Mac-Rae commented:
Assessed by Travis :+1:

Dillon-Brown commented:
Interesting that this passed the acceptance tests.... It looks like you'd also need to revert the change to ```Localization.php``` as the config hasn't been loaded when it attempts to get the DB instance causing it to fatal.

dfstrauss commented:
@Dillon-Brown My change takes that into account.
Previously the function exited early if `$db` was not yet initialized:
```
global $db;
global $sugar_config;
if (empty($db)) {
    return array();
}
```
I changed it to exit early when `$sugar_config` is not yet set, achieving the same thing.

My original change to `Localization.php` caused installation to fail, until I moved the
```
$db = DBManagerFactory::getInstance();
```
line to after that check.

dfstrauss commented:
@Dillon-Brown I fixed it now. The problem is that during install, the db config is not yet available. My original check (`if (empty($sugar_config))`) works if there's no `config_override.php` file but fails if there is one.
I changed it now to `if (empty($sugar_config['dbconfig']))`
